src/redisvalue.rs

```diff
@@ -102,15 +150,30 @@ impl<T: Into<Self>> From<Vec<T>> for RedisValue {
     }
 }
 
-impl<T: CallReply> From<&T> for RedisValue {
-    fn from(reply: &T) -> Self {
-        match reply.get_type() {
-            raw::ReplyType::Error => RedisValue::Error(reply.get_string().unwrap()),
-            raw::ReplyType::Unknown => RedisValue::StaticError("Error on method call"),
-            raw::ReplyType::Array => RedisValue::Array(reply.iter().map(|v| (&v).into()).collect()),
-            raw::ReplyType::Integer => RedisValue::Integer(reply.get_int()),
-            raw::ReplyType::String => RedisValue::SimpleString(reply.get_string().unwrap()),
-            raw::ReplyType::Null => RedisValue::Null,
+impl<'root> From<&CallReply<'root>> for RedisValue {
+    fn from(reply: &CallReply<'root>) -> Self {
+        match reply {
+            CallReply::Error(reply) => RedisValue::Error(reply.to_string().unwrap()),
+            CallReply::Unknown => RedisValue::StaticError("Error on method call"),
+            CallReply::Array(reply) => {
+                RedisValue::Array(reply.iter().map(|v| (&v).into()).collect())
+            }
+            CallReply::I64(reply) => RedisValue::Integer(reply.to_i64()),
+            CallReply::String(reply) => RedisValue::SimpleString(reply.to_string().unwrap()),
+            CallReply::Null(_) => RedisValue::Null,
+            CallReply::Map(reply) => {
+                RedisValue::Map(reply.iter().fold(HashMap::new(), |mut acc, (key, val)| {
```

Comment:
I did not implement into iter and we can not really implement it because the inner element can not outlive the futher.

Reply:
@vityafx I do not see how you example demonstrate the use of `into_iter`, it uses `iter` which is what I have implemented.
There is not point in consuming the root because if the root is freed the you can not hold the children...